libs/testserver/server.go

```diff
@@ -34,7 +34,8 @@ type Request struct {
 	Headers http.Header `json:"headers,omitempty"`
 	Method  string      `json:"method"`
 	Path    string      `json:"path"`
-	Body    any         `json:"body"`
+	Body    any         `json:"body,omitempty"`
+	RawBody string      `json:"raw_body,omitempty"`
```

Comment:
This is addressed in https://github.com/databricks/cli/pull/2285/files. We don't need a separate `raw_body` field.

Reply:
Do you depend on having single 'body' field somewhere? 

I went with separate fields because they will always have different types. Body is likely to be object and RawBody is a string. So filtering changes. 

Presence of "raw_body" is also a good signal to the reviewer that body is not JSON. You don't get that clarity with one field.

Reply:
Regarding your PR - I think it should be split, "Add acceptance tests for auth resolution" and fixing this bug are quite different.

Reply:
> Do you depend on having single 'body' field somewhere?

No, but having a single body field mirrors a "real" HTTP request. For example, If we start recording requests to generate acceptance tests, then not having the `raw_body` here would be marginally better.

> Presence of "raw_body" is also a good signal to the reviewer that body is not JSON. You don't get that clarity with one field.

You do, right? In `out.requests.txt,` it's visible whether the body is a JSON object or a non-JSON string.

> I went with separate fields because they will always have different types. Body is likely to be object and RawBody is a string. So filtering changes.

What do you mean by filtering changes?

Reply:
Discussed offline, we decided to go with the existing approach.